File: solutions/BearPaints/solve/python/BearPaintsSolver.py
#! /usr/bin/env python
import os, sys
import BearPaints

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)
    
        W = reader.next("int")
        reader.next()
        H = reader.next("int")
        reader.next()
        M = reader.next("long")

    result = BearPaints.maxArea(W, H, M)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "long"))

if __name__ == "__main__":
    init()
    main()

File: solutions/BearPaints/solve/ruby/BearPaintsSolver.rb
#! /usr/bin/env ruby

require_relative "BearPaints"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    W = reader.next(TInt)
    reader.next()
    H = reader.next(TInt)
    reader.next()
    M = reader.next(TLong)

    result = BearPaints.new().maxArea(W, H, M)
    IO.write(ARGV[1], Writer.new().next(result, TLong).to_s())
end
init()
main()
